File: croisé/AFTAS-backend/src/main/java/com/api/AFTAS/domains/competition/Competition.java
package com.api.AFTAS.domains.competition;

import com.api.AFTAS.domains.hunting.Hunting;
import com.api.AFTAS.domains.ranking.Ranking;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
public class Competition {
    @Id
    private String code;
    private LocalDate date;
    private LocalTime startTime;
    private LocalTime endTime;
    private Integer numberOfParticipants;
    private String location;
    private Double amount;
    @OneToMany(mappedBy = "competition")
    private List<Hunting> huntings = new ArrayList<>();
    @OneToMany(mappedBy = "id.competition")
    private List<Ranking> rankings;
}

File: WI-GOLD-FISH-FRONT/src/app/model/Competition.ts
import { Time } from "@angular/common";
import Hunting from "./Hunting";
import Ranking from "./Ranking";

export default interface Competition {
    code : string;
    date : Date;
    startTime : Time | string;
    endTime : Time | string;
    numberOfParticipants : number;
    location : string;
    amount : number;
    huntings : Hunting[];
    ranking : Ranking[];
}